ExtractorUtils/Configuration/BaseConfig.cs

```diff
@@ -48,6 +48,11 @@ public override void GenerateDefaults()
             if (Logger == null) Logger = new LoggerConfig();
             if (StateStore == null) StateStore = new StateStoreConfig();
         }
+
+        /// <summary>
+        /// Configuration for automatically reporting extraction pipeline runs.
+        /// </summary>
+        public ExtractionRunConfig ExtractionPipeline { get; set; }
```

Comment:
I was thinking to have this in the cognite section, and not as its own section, so
``` yaml
cognite:
    ...
    extraction-pipeline:
        id: 123
        # or
        external-id: abc
```
How are data sets typically configured for pi/opc ua?

Reply:
There isn't really a need to have `id` in there, since pipelines require external id.

I still haven't found a good way to handle dataset externalId/internalId, most resources require internal id, but we usually support externalId.

Reply:
I agree it should probably be in the cognite section, there were some issues with the dependency injection, but I'll figure it out.

Reply:
Internal ID is true, didn't think of that. I thought I would reuse the new way of configuring data sets in the python utils for consistency, even though using external id is enough for extpipes (you actually need to use external id when reporting runs iirc, so for extpipes the issue is opposite that of data sets). 

This is completely on the side, but for the internalid/externalid issue with datasets - could we do something similar to the python one where we resolve the data set at startup?

Reply:
We could, adding a "DataSet" abstraction could be useful. That said, you still have to reference that on your own when creating assets/timeseries.

Reply:
I'll add it to the list eventually...